core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -969,6 +970,71 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     }
   }
 
+  @Test
+  def testCoopearativeAssigorAndRejoin(): Unit = {
```

Comment:
1. Maybe rename to `testRebalanceAndRejoin`
2. Could we also test eager assignor? Ex:
```
  @ParameterizedTest
  @ValueSource(strings = Array(classOf[CooperativeStickyAssignor].getName, classOf[RangeAssignor].getName))
  def testRebalanceAndRejoin(partitionAssignmentStrategy: String): Unit = {
```
I think the only place we need to change is the last generationId should be +1 or +2, right?